### PR TITLE
Allow fake IP works for all IPs

### PR DESCRIPTION
**Pull request by nouchka**
when local_ip is empty, allow to change all ip with fake one. usefull when we can't predict dev IP when using containers for example

### CODE REVIEW

File: Plugin/FakeIpPlugin.php [hunk 1]
```diff
@@ -75,6 +75,8 @@ public function handleQuery(Query $query, callable $next, callable $first)
         $text = str_replace($this->needle, $replacement, $query->getText(), $count);
         if ($count > 0) {
             $query = $query->withText($text);
+        } else if (empty($this->needle)) {
```
Comment (norkunas) [inline Plugin/FakeIpPlugin.php:78]:
```suggestion
        } else if ($this->needle !== null && $this->needle !== '') {
```

**Comment by norkunas** [comment]
Sorry, forgot that Symfony cs rules uses yoda style :) could you fix cs-fixer warnings?

File: Plugin/FakeIpPlugin.php [hunk 2]
```diff
@@ -76,6 +76,9 @@ public function handleQuery(Query $query, callable $next, callable $first)
         if ($count > 0) {
             $query = $query->withText($text);
         }
+        if (null === $this->needle || '' === $this->needle) {
+            $query = $query->withText($this->replacement);
```
Comment (norkunas) [inline Plugin/FakeIpPlugin.php:80]:
```suggestion
            $query = $query->withText($replacement);
```
because it could be generated from faker

**Comment by norkunas** [comment]
and please add a unit test :wink:

File: Plugin/FakeIpPlugin.php (hunk 2 above)
Comment (nouchka) [reply Plugin/FakeIpPlugin.php:80]: oups sorry, I will update this

File: Tests/Plugin/FakeIpPluginTest.php [hunk 3]
```diff
@@ -33,6 +33,17 @@ public function testSimpleHandleQuery()
         $this->assertSame($query->getText(), '123.123.123.123');
     }
 
+    public function testAllIpQuery()
```
Comment (norkunas) [inline Tests/Plugin/FakeIpPluginTest.php:36]: Maybe add a data provider to test both null and empty string cases? :)

File: Tests/Plugin/FakeIpPluginTest.php [hunk 4]
```diff
@@ -33,6 +33,28 @@ public function testSimpleHandleQuery()
...
+    {
+        $fakeIpPlugin = new FakeIpPlugin(null, '123.123.123.123');
+        $query = GeocodeQuery::create('124.124.124.124');
+
+        /** @var Query $query */
+        $query = $fakeIpPlugin->handleQuery($query, function (Query $query) { return $query; }, function () {});
+
+        $this->assertSame($query->getText(), '123.123.123.123');
+    }
+
```
Comment (norkunas) [inline Tests/Plugin/FakeIpPluginTest.php:57]:
```suggestion
    /**
     * @testWith [null]
     *                   ['']
     */
    public function testEmptyLocalIpQuery(?string $localIp)
    {
        $fakeIpPlugin = new FakeIpPlugin($localIp, '123.123.123.123');
        $query = GeocodeQuery::create('124.124.124.124');

        /** @var Query $query */
        $query = $fakeIpPlugin->handleQuery($query, function (Query $query) { return $query; }, function () {});

        $this->assertSame($query->getText(), '123.123.123.123');
    }
```
try this :)

File: Tests/Plugin/FakeIpPluginTest.php [hunk 5]
```diff
@@ -33,6 +33,20 @@ public function testSimpleHandleQuery()
...
+    {
+        $fakeIpPlugin = new FakeIpPlugin($localIp, '123.123.123.123');
+        $query = GeocodeQuery::create('124.124.124.124');
+
+        /** @var Query $query */
+        $query = $fakeIpPlugin->handleQuery($query, function (Query $query) { return $query; }, function () {});
+
+        $this->assertSame($query->getText(), '123.123.123.123');
+    }
     public function testHandleQueryUsingFaker()
```
Comment (norkunas) [inline Tests/Plugin/FakeIpPluginTest.php:50]:
```suggestion
    }

    public function testHandleQueryUsingFaker()
```

**Comment by norkunas** [comment]
Thank you @nouchka :)

**Comment by nouchka** [comment]
Thank you too :)
